addons/project/static/src/components/project_task_state_selection/project_task_stage_state_selection/project_task_stage_with_state_selection.js

```diff
@@ -0,0 +1,62 @@
+import { _t } from "@web/core/l10n/translation";
+import { standardFieldProps } from "@web/views/fields/standard_field_props";
+import { Many2OneField } from "@web/views/fields/many2one/many2one_field";
+
+import { registry } from "@web/core/registry";
+import { Component } from "@odoo/owl";
+import { omit } from "@web/core/utils/objects";
+
+import { ProjectTaskStateSelection } from "../project_task_state_selection";
+
+
+export class TaskStageWithStateSelection extends Component {
+    static template = "project.TaskStageWithStateSelection";
+
+    static props = {
+        ...standardFieldProps,
+        stateReadonly: { type: Boolean, optional: true },
+        viewType: { type: String },
+    };
+
+    static components = {
+        ProjectTaskStateSelection,
+        Many2OneField,
+    };
+
+    get stageProps() {
+        return omit(this.props, "stateReadonly", "viewType");
+    }
+
+    get stateProps() {
+        return {
+            ...this.stageProps,
+            name: "state",
+            readonly: this.props.stateReadonly,
+            viewType: this.props.viewType,
+            showLabel: false,
+        };
+    }
+}
+
+export const taskStageWithStateSelection = {
+    component: TaskStageWithStateSelection,
+    supportedOptions: [
+        {
+            label: _t("State readonly"),
+            name: "state_readonly",
+            type: "boolean",
+            default: true,
+            help: _t("If set, state is set to readonly."),
```

Comment:
I don't think it is needed to add helper here, the label is clear enough, IMO. 😇

Reply:
Done